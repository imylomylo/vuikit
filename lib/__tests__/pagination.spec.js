import { createVue, destroyVM, getSnapshot } from './_util'
// import { toArray } from 'helpers/util'
import waitForUpdate from './_wait-for-update'

describe('Pagination', () => {
  let vm
  let snapshot
  afterEach(() => {
    destroyVM(vm)
  })

  fit('renders correctly', () => {
    snapshot = getSnapshot(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="1" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(snapshot).toMatchSnapshot()
    destroyVM(vm)

    snapshot = getSnapshot(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="10" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(snapshot).toMatchSnapshot()
    destroyVM(vm)

    snapshot = getSnapshot(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="20" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(snapshot).toMatchSnapshot()
    destroyVM(vm)

    snapshot = getSnapshot(`
      <vk-pagination :limit="10" :active="10" :total="200">
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(snapshot).toMatchSnapshot()
    destroyVM(vm)
  })

  it.skip('triggers event change when a not active page is clicked', done => {
    vm = createVue(`
      <vk-pagination>
        <vk-pagination-prev />
        <vk-pagination-pages :active="20" :total="200" :limit="10" />
      </vk-pagination>
    `, true)

    const cb = jest.fn()
    const pagination = vm.$children[0]

    pagination.$on('change', cb)
    vm.$el.querySelector('li a').click()
    waitForUpdate(() => {
      const args = cb.mock.calls[0]
      const { page } = args[0]
      expect(cb).toHaveBeenCalled()
      expect(args).toHaveLength(1)
      expect(page).toEqual(20)
    }).then(done)
  })
})
