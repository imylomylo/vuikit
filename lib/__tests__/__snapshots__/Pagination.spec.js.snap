exports[`Pagination renders correctly 1`] = `
"html
  body
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li.uk-active
        span 1
      li
        a 2
      li
        a 3
      li
        a 4
      li
        a 5
      li
        a 6
      li
        a 7
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
"
`;

exports[`Pagination renders correctly 2`] = `
"html
  body
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li.uk-active
        span 1
      li
        a 2
      li
        a 3
      li
        a 4
      li
        a 5
      li
        a 6
      li
        a 7
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a 1
      li
        span ...
      li
        a 7
      li
        a 8
      li
        a 9
      li.uk-active
        span 10
      li
        a 11
      li
        a 12
      li
        a 13
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
"
`;

exports[`Pagination renders correctly 3`] = `
"html
  body
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li.uk-active
        span 1
      li
        a 2
      li
        a 3
      li
        a 4
      li
        a 5
      li
        a 6
      li
        a 7
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a 1
      li
        span ...
      li
        a 7
      li
        a 8
      li
        a 9
      li.uk-active
        span 10
      li
        a 11
      li
        a 12
      li
        a 13
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a 1
      li
        span ...
      li
        a 14
      li
        a 15
      li
        a 16
      li
        a 17
      li
        a 18
      li
        a 19
      li.uk-active
        span 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
"
`;

exports[`Pagination renders correctly 4`] = `
"html
  body
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li.uk-active
        span 1
      li
        a 2
      li
        a 3
      li
        a 4
      li
        a 5
      li
        a 6
      li
        a 7
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a 1
      li
        span ...
      li
        a 7
      li
        a 8
      li
        a 9
      li.uk-active
        span 10
      li
        a 11
      li
        a 12
      li
        a 13
      li
        span ...
      li
        a 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a 1
      li
        span ...
      li
        a 14
      li
        a 15
      li
        a 16
      li
        a 17
      li
        a 18
      li
        a 19
      li.uk-active
        span 20
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
    ul.uk-pagination
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-prev.uk-icon(uk-pagination-previous=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
      |  
      li
        a
          span.uk-pagination-next.uk-icon(uk-pagination-next=\'true\')
"
`;
